proc/threads_windows.go

```diff
@@ -0,0 +1,142 @@
+package proc
+
+// #include "threads_windows.h"
+import "C"
+import (
+	"fmt"
+	"unsafe"
+	"syscall"
+	sys "golang.org/x/sys/windows"
+)
+
+type WaitStatus sys.WaitStatus
+
+type OSSpecificDetails struct {
+	hThread syscall.Handle
+}
+
+func (t *Thread) halt() (err error) {
+	// TODO: We are currently ignoring the request to halt.
+	// I believe this is guaranteed to already be halted. 
+	return nil
+}
+
+func (t *Thread) singleStep() error {	
+	var context C.CONTEXT
+	context.ContextFlags = C.CONTEXT_ALL;
+	
+	res, err := C.GetThreadContext(C.HANDLE(unsafe.Pointer(t.os.hThread)), &context)
+	if res == 0 {
+		return err
+	}
+	
+	//TODO: Is it really okay to not decrement to IP?
+	//context.Rip--;
+	context.EFlags |= 0x100;
+	
+	
+	res, err = C.SetThreadContext(C.HANDLE(unsafe.Pointer(t.os.hThread)), &context)
+	if res == 0 {
+		return err
+	}
+	
+	// TODO: It's possible that this resume/trapWait will 
+	// land us on another thread. We need to guard for that 
+	// and loop here till we sucessfully land back on our thread.
+	err = t.resume()
+	if err != nil {
+		return err
+	}
+	_, err = t.dbp.trapWait(0)
```

Comment:
That's true. @aarzilli is there a test for that? If not we should add one, because for now let's simply not use trapWait here, but shortly after this and some other PRs land I'd like to look into ensuring that trapWait is safe to call in that situation. Either way I'd like to prevent that as a regression in the future.